src/agg/process_markers_symbolizer.cpp

```diff
@@ -107,22 +107,26 @@ struct agg_markers_renderer_context : markers_renderer_context
                 double sample_y = std::floor(dy * sampling_rate);
 
                 int sample_idx = static_cast<int>(sample_y) * sampling_rate + static_cast<int>(sample_x);
-
                 std::tuple<svg_path_ptr, int, svg::path_attributes> key(src, sample_idx, attrs[0]);
 
+                std::shared_ptr<image_rgba8> fill_img = nullptr;
+                std::shared_ptr<image_rgba8> stroke_img = nullptr;
+                bool cache_hit = false;
+                // Limit the scope of the cache mutex
+                {
```

Comment:
I planned to add the config flag check here as well, but doubted about locking/unlocking several times. I believe you tested performance of it so.. :+1: